### PR TITLE
The Seller and User -  list products

### PR DESCRIPTION
**Pull request by ndahimana154**

### CODE REVIEW

**Review by ProgrammerDATCH — CHANGES_REQUESTED** [review]
You mixed middlewares with Controller.

**Review by Jadowacu1 — DISMISSED** [review]

**Review by MANISHIMWESalton — DISMISSED** [review]

**Review by Jadowacu1 — DISMISSED** [review]

File: src/middlewares/validation.ts [hunk 1]
```diff
@@ -276,4 +277,66 @@ const isGoogleEnabled = async (req: any, res: Response, next: NextFunction) => {
...
\ No newline at end of file
+const productsByCategory = async (req: Request, res: Response, next: NextFunction) => {
+    const category = req.query.category
+    if (category) {
+        const products = await productRepositories.getAvailableProductsByAttributes("category", category)
+        return res.status(httpStatus.OK).json({ status: httpStatus.OK, data: products })
+    }
+    return next()
+}
+const sellersShop = async (req: any, res: Response, next: NextFunction) => {
```
Comment (AimePazzo) [inline src/middlewares/validation.ts:288]: this function middleware is the same as `IsSellerShopExist` just use one

File: src/middlewares/validation.ts [hunk 2]
```diff
@@ -276,4 +277,66 @@ const isGoogleEnabled = async (req: any, res: Response, next: NextFunction) => {
...
+        if (!shop) {
+            return res.status(httpStatus.NOT_FOUND).json({ status: httpStatus.NOT_FOUND, message: "Shop not found" });
+        }
+        req.shop = shop;
+        return next();
+    } catch (error) {
+        return res.status(httpStatus.INTERNAL_SERVER_ERROR).json({ status: httpStatus.INTERNAL_SERVER_ERROR, message: error.message });
+    }
+}
+const isProductsByCategorySelected = async (req: Request, res: Response, next: NextFunction) => {
```
Comment (AimePazzo) [inline src/middlewares/validation.ts:316]: Also, 'isProductsbyCategorySelected' is essentially the same as 'productsByCategory'. It seems like you are duplicating functions.

**Review by Jadowacu1 — DISMISSED** [review]

**Review by MANISHIMWESalton — DISMISSED** [review]

**Review by Jadowacu1 — APPROVED** [review]

**Review by SaddockAime — DISMISSED** [review]

**Review by Jadowacu1 — DISMISSED** [review]

**Review by AimePazzo — APPROVED** [review]

**Review by ProgrammerDATCH — DISMISSED** [review]

**Review by solangeihirwe03 — DISMISSED** [review]

**Review by MANISHIMWESalton — DISMISSED** [review]

**Review by Y-elv — DISMISSED** [review]
you did good

**Review by SaddockAime — DISMISSED** [review]

**Review by Jadowacu1 — DISMISSED** [review]

**Review by MANISHIMWESalton — DISMISSED** [review]

**Review by AimePazzo — APPROVED** [review]

**Review by SaddockAime — DISMISSED** [review]

**Review by Jadowacu1 — DISMISSED** [review]

**Review by Jadowacu1 — DISMISSED** [review]

**Review by Jadowacu1 — APPROVED** [review]

**Review by MANISHIMWESalton — APPROVED** [review]

**Review by hbapte — APPROVED** [review]

**Pull request merged**